File: problems/week01/SWEA_2001/YeoroJeong/2001.py
T = int(input())

for test_case in range(1, T + 1):
    N, M = map(int, input().split())

    # N × N 크기의 파리 개수 배열 입력
    arr = [list(map(int, input().split())) for _ in range(N)]

    # 한 번에 잡을 수 있는 최대 파리 수
    answer = 0

    # M × M 파리채의 왼쪽 위 시작 위치를 모두 탐색
    # 시작 인덱스는 0부터 N-M까지 가능하므로 N-M+1번 반복
    for i in range(N - M + 1):
        for j in range(N - M + 1):
            temp_sum = 0

            # 현재 시작 위치 (i, j)부터 M × M 영역의 합 계산
            for row in range(i, i + M):
                for col in range(j, j + M):
                    temp_sum += arr[row][col]

            # 지금까지 계산한 파리 수 중 최댓값 갱신
            answer = max(answer, temp_sum)

    print(f"#{test_case} {answer}")

File: problems/week01/SWEA_2001/DaewonKim/2001.cpp
/*

*/
#include <iostream>
#include <vector>

using namespace std;

int main(int argc, char** argv)
{
	int test_case;
	int T;
	
	cin>>T;

	for(test_case = 1; test_case <= T; ++test_case)
	{

        int n,m;
        cin >> n;
        cin >> m;

        vector<vector<int>> board(n, vector<int> (n,0));
        for(int row = 0; row<n; ++row)
        {
            for(int col = 0; col<n; ++col)
            {
                cin >> board[row][col];
            }
        }


        int maxFly = 0;
        // 크기가 m x m인 것을 만들기
        for(int row = 0; row<n; ++row)
        {
            for(int col = 0; col<n; ++col)
            {
                
                int endRow = row+(m-1); // 열(세로)
                int endCol = col+(m-1); // 행(가로)
                if(endRow < n && endCol < n ) // 끝이 범위 안이라면
                {
                    int tmpFly=0;
                    for(int curRow = row; curRow<=endRow; ++curRow)
                    {
                        for(int curCol = col; curCol<=endCol; ++curCol)
                        {
                            tmpFly += (board[curRow][curCol]);
                        }
                    }
                    maxFly = max(maxFly, tmpFly);
                }
                else{
                    continue;
                }


            }
        }

        cout << "#" << test_case << " " << maxFly << "\n";
	}
	return 0;//정상종료시 반드시 0을 리턴해야합니다.
}